lib/processor/validate.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const Q = require('q');
+const fs = require('fs');
+const ffmpeg = require('fluent-ffmpeg');
+
+// non-fatal ffprobe errors
+const KNOWN_ERRORS = [
+  /invalid data found/i,
+  /invalid argument/i,
+  /misdetection possible/i
+];
+
+// identify and validate a file
+module.exports = (filePath, ffprobePromise) => {
+  if (ffprobePromise === undefined) {
+    ffprobePromise = Q.ninvoke(ffmpeg(filePath), 'ffprobe');
+  }
+
+  return ffprobePromise.then(
+    info => {
+      let meta = {size: info.format.size, hasAudio: false, hasVideo: false};
+      let audioStream = info.streams.find(s => s.codec_type === 'audio');
+      let videoStream = info.streams.find(s => s.codec_type === 'video' && s.duration > 0);
+
+      // audio stream info
+      if (audioStream) {
```

Comment:
before there was another check for mp2/mp3 in addition to checking for codec_type 'audio' -- no longer necessary?

Reply:
Yeah, I think we're moving that logic out of the lambda - it's responsible for returning info, but not making any judgements based on that.

Reply:
👍